### ISSUE TITLE
aws_ses_receipt_rule - Invalid KMS encryption key ARN

### ISSUE THREAD

**Issue by melbit-anthonyw** [terraform-providers/terraform-provider-aws#4906]
### Terraform Version

0.11.3

### Affected Resource(s)

aws_ses_receipt_rule

### Terraform Configuration Files

```hcl
resource "aws_ses_receipt_rule" "ses_resource" {
  name          = "${var.account}-deploy"
  rule_set_name = "default-rule-set"
  recipients    = ["${var.account}-deploy@blah.com.au"]
  enabled       = true
  scan_enabled  = true
  provider      = "aws.blah-us-east"

  s3_action {
    bucket_name       = "ses-mail"
    object_key_prefix = "${var.account}-deploy"
    position          = 1
  }
}
```

### Debug Output

2018-06-21T16:30:56.428+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: 2018/06/21 16:30:56 [DEBUG] [aws-sdk-go] DEBUG: Response email/CreateReceiptRule Details:
2018-06-21T16:30:56.428+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: ---[ RESPONSE ]--------------------------------------
2018-06-21T16:30:56.428+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: HTTP/1.1 400 Bad Request
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: Connection: close
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: Content-Length: 284
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: Content-Type: text/xml
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: Date: Thu, 21 Jun 2018 06:30:56 GMT
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: X-Amzn-Requestid: a7692543-751c-11e8-8518-97ae23c79dbf
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4:
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4:
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: -----------------------------------------------------
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: 2018/06/21 16:30:56 [DEBUG] [aws-sdk-go] <ErrorResponse xmlns="http://ses.amazonaws.com/doc/2010-12-01/">
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4:   <Error>
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4:     <Type>Sender</Type>
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4:     <Code>InvalidParameterValue</Code>
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4:     <Message>Invalid KMS encryption key ARN: </Message>
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4:   </Error>
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4:   <RequestId>a7692543-751c-11e8-8518-97ae23c79dbf</RequestId>
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: </ErrorResponse>
2018-06-21T16:30:56.429+1000 [DEBUG] plugin.terraform-provider-aws_v1.19.0_x4: 2018/06/21 16:30:56 [DEBUG] [aws-sdk-go] DEBUG: Validate Response email/CreateReceiptRule failed, not retrying, error InvalidParameterValue: Invalid KMS encryption key ARN:

### Expected Behavior

It should be creating the SES rule

### Actual Behavior

 Error creating SES rule: InvalidParameterValue: Invalid KMS encryption key ARN

Fails with reference to KMS ARN being invalid even though it's an optional parameter only used if encryption is enabled

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Previous runs back on the 15th of March were successful. No code changes and the following block

```hcl
  enabled:                                "" => "true"
  name:                                   "" => "blah-deploy"
  recipients.#:                           "" => "1"
  recipients.689644031:                   "" => "blah-deploy@blah.com.au"
  rule_set_name:                          "" => "default-rule-set"
  s3_action.#:                            "" => "1"
  s3_action.3142081338.bucket_name:       "" => "ses-mail"
  s3_action.3142081338.kms_key_arn:       "" => ""
  s3_action.3142081338.object_key_prefix: "" => "blah-deploy"
  s3_action.3142081338.position:          "" => "1"
  s3_action.3142081338.topic_arn:         "" => ""
  scan_enabled:                           "" => "true"
  tls_policy:                             "" => "<computed>"[0m
```

However the exact same code run today fails with the above error. I thought it maybe a new provider was released so tried pinning the version back to what was used when the code was run on the 15th (1.19) but it didn't fix the problem.

**Comment by melbit-anthonyw** [comment]
Further investigation looks like there was a KMS key created in my account automatically by AWS on the 21/06/2018 specifically for SES (aws/ses) with the description 

> Default master key that protects my SES emails when no other key is defined

When done through the GUI and I select encryption it automatically selects the key but it still shouldn't be attempting to use the key if encryption isn't selected... at least logically anyway. Looks like AWS may have changed/screwed something with the API? Specifying the ARN to the default key works (Or any key in that region for that matter).

**Comment by saravanan30erd** [comment]
submitted PR: #4965

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.25.0 of the AWS provider, likely middle of this week.

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
